test/e2e/api_open_banking_test.go

```diff
@@ -0,0 +1,324 @@
+					ApplicationName:   "test",
+					ClientRedirectURL: "https://www.google.com",
+				},
+			)
+			Expect(err).To(BeNil())
+			Expect(resp.GetV3PaymentServiceUserCreateLinkResponse().GetAttemptID()).To(Not(BeEmpty()))
+			Expect(resp.GetV3PaymentServiceUserCreateLinkResponse().GetLink()).To(Not(BeEmpty()))
+
+			link, err := url.Parse(resp.GetV3PaymentServiceUserCreateLinkResponse().GetLink())
+			Expect(err).To(BeNil())
+
+			appUrl, err := url.Parse(app.GetValue().URL())
+			Expect(err).To(BeNil())
+
+			// Here, we don't care about the link sent back from dummypay, we
+			// just want to validate that the redirect endpoint is doing its job
+			// when called, so we can actually replace the host and path to
+			// avoid creating a gateway inside the test suite.
+			link.Scheme = appUrl.Scheme
+			link.Host = appUrl.Host
+			link.Path = strings.TrimPrefix(link.Path, "api/payments/")
+
+			req, err := http.NewRequestWithContext(ctx, http.MethodPost, link.String(), nil)
+			Expect(err).To(BeNil())
+
+			statusCode, err := httpClient.Do(ctx, req, nil, nil)
+			Expect(err).To(BeNil())
+			Expect(statusCode).To(Equal(http.StatusNoContent))
+
+			attemptPoller := pollAttempts(ctx, app, psuID, connectorID, GinkgoT())
+			Eventually(attemptPoller()).WithTimeout(10 * time.Second).Should(HaveLinkAttemptsLengthMatcher(1, []PayloadMatcher{HaveLinkAttemptStatus(components.V3PSUBankBridgeConnectionAttemptStatusEnumCompleted)}...))
+
+			connectionPoller := pollConnection(ctx, app, psuID, GinkgoT())
+			Eventually(connectionPoller()).WithTimeout(10 * time.Second).Should(HaveUserConnectionsLengthMatcher(1, []PayloadMatcher{HaveUserConnectionStatus(components.V3ConnectionStatusEnumActive)}...))
+
+			// We should be able to delete the connection
+			connectionID := connectionPoller()()[0].ConnectionID
+			_, err = app.GetValue().SDK().Payments.V3.DeletePaymentServiceUserConnectionFromConnectorID(ctx, psuID, connectorID, connectionID)
+			Expect(err).To(BeNil())
+			Eventually(connectionPoller()).WithTimeout(10 * time.Second).Should(HaveUserConnectionsLengthMatcher(0))
+		})
+	})
+
+	When("creating a link and call it - error", func() {
+		var (
+			connectorID string
+			httpClient  httpwrapper.Client
+		)
+
+		BeforeEach(func() {
+			httpClient = httpwrapper.NewClient(&httpwrapper.Config{})
+
+			var err error
+
+			id := uuid.New()
+			conf := newV3ConnectorConfigFn()(id)
+			conf.LinkFlowError = pointer.For(true)
+			conf.UpdateLinkFlowError = pointer.For(false)
+			connectorID, err = installV3Connector(ctx, app.GetValue(), conf, id)
+			Expect(err).To(BeNil())
+
+			forwardResponse, err := app.GetValue().SDK().Payments.V3.ForwardPaymentServiceUserToBankBridge(
+				ctx,
+				psuID,
+				connectorID,
+			)
+			Expect(err).To(BeNil())
+			Expect(forwardResponse.GetHTTPMeta().Response.StatusCode).To(Equal(http.StatusNoContent))
+		})
+
+		AfterEach(func() {
+			uninstallConnector(ctx, app.GetValue(), connectorID)
+		})
+
+		It("should fail", func() {
```

Comment:
the diff between here and the success case is not super obvious to find, anything we can do better?